gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HookHandler.java

```diff
@@ -1030,16 +1032,24 @@ private void handleListenerRegistration(final HttpServerRequest request) {
             final String listenerStorageUri = hookRootUri + HOOK_LISTENER_STORAGE_PATH + getUniqueListenerId(request.uri());
 
             // Extract expireAfter from the registration header.
-            Integer expireAfter = ExpiryCheckHandler.getExpireAfter(request.headers());
-            if (expireAfter == null) {
-                expireAfter = DEFAULT_HOOK_STORAGE_EXPIRE_AFTER_TIME;
-            }
+            final String myExpireAfter = request.headers().get( X_EXPIRE_AFTER );
+            final String expirationTime;
+            if( "-1".equals(myExpireAfter) ){
```

Comment:
Why not add this to the ExpiryCheckHandler where we already provide methods for expiry headers? You could provide a method which returns an enum with values like **NEVER**, **EXPIRES_IN_MS** holding the corresponding Integer values. The ExpiryCheckHandler already handles the "-1" case by returning null. You could not return null in this case but the Expiration.NEVER enum value.